fdbctl/protos/control_service.proto

```diff
@@ -2,5 +2,399 @@ syntax = "proto3";
+        // Datacenter identifier - identifies which datacenter this worker is in
+        optional string dc_id = 4;
+
+        // Data hall identifier - identifies which data hall within a datacenter
+        optional string data_hall_id = 5;
+    }
+
+    // Network address where this worker can be reached (e.g., "127.0.0.1:4500")
+    optional string address = 1;
+
+    // gRPC address for this worker if gRPC is enabled
+    optional string grpc_address = 2;
+
+    // Process class determines the worker's role (e.g., "storage", "transaction", "stateless")
+    optional string process_class = 3;
+
+    // Locality information describing where this worker is physically/logically located
+    optional Locality locality = 4;
+}
+
+// Request to retrieve the current cluster coordinators.
+// This returns the list of processes that are currently acting as coordinators
+// for the cluster's coordination state.
+message GetCoordinatorsRequest {}
+
+// Response containing the current cluster coordinators.
+message GetCoordinatorsReply {
+    // List of coordinator addresses in the format "ip:port"
+    repeated string coordinators = 1;
+}
+
+// Request to change the cluster's coordinators.
+// Coordinators maintain the cluster's configuration and state. Changing them
+// is a critical operation that should be done carefully.
+message ChangeCoordinatorsRequest {
+    // Human-readable description for the cluster (e.g., cluster name)
+    optional string cluster_description = 1;
+
+    // If true, disables the configuration database
+    optional bool disable_config_db = 2;
+
+    // If true, automatically selects coordinators based on the current cluster topology.
+    // When false, uses the addresses specified in new_coordinator_addresses.
+    optional bool automatic_coordinators = 3;
+
+    // List of addresses to use as new coordinators (when automatic_coordinators is false).
+    // Each address should be in the format "ip:port".
+    repeated string new_coordinator_addresses = 4;
+}
+
+// Response to a coordinator change operation.
+message ChangeCoordinatorsReply {
+    // True if the coordinators were actually changed, false if they remained the same
+    optional bool changed = 1;
+
+    // The current list of coordinator addresses after the operation
+    repeated string coordinators = 2;
+}
+
+// Request for automatic configuration suggestions.
+// Analyzes the current cluster and returns suggested configuration parameters.
+message ConfigureAutoSuggestRequest {}
+
+// Response containing suggested configuration parameters.
+message ConfigureAutoSuggestReply {
+    // Suggested configuration that can be used with the Configure RPC
+    ConfigureRequest configure_request = 1;
+}
+
+// Request to configure database settings.
+// This is one of the most critical operations for a FoundationDB cluster,
+// affecting redundancy, storage engines, process roles, and various other settings.
+message ConfigureRequest {
+    // Database creation flags
```

Comment:
Could we add some additional explanation what this means? Is the expectation that all those flags are provided when the initial database is created?